### ISSUE TITLE
Get duration of nested items in composed audio

### ISSUE THREAD

**Issue by ryanheise** [ryanheise/just_audio#141]
Note to self: Provide a way to query any nested audio source for its duration. E.g. items within a playlist.

**Comment by ReaganRealones** [comment]
Exactly, I was actually going to submit a new issue requesting for this support. Please let me know as soon as you are done so I try it out. Thank you @ryanheise  👍

**Comment by ryanheise** [comment]
Will try, although not sure how long it will take me to get around to it! I seem to have quite a backlog of issues right now :-)

**Comment by ryanheise** [comment]
FYI, the latest release adds a `duration` property of `IndexedAudioSource` which partially implements this feature request. The catch is that the durations of nested audio sources aren't set until the player reaches that item by advancing or seeking to it. This could be improved in some special cases.

Note that it is also possible for an application to manually set the duration property on each `IndexedAudioSource` if it is already known (which in certain applications it is.)

**Comment by rupakswadesh** [comment]
when seek to next perform then not update duration issue? i am getting previous song duration why?

**Comment by ryanheise** [comment]
Hi @rupakswadesh if you have found a bug, you will need to open a separate bug report. I noticed you already posted another issue, but that was not a valid bug report because you completely ignored and deleted all of the instructions about what you were required to do.

**Comment by phamconganh** [comment]
@ryanheise in your `example_playlist`. When I remove the `duration` property in `MediaItem` of `MediaLibrary`. Can I get the duration of current play audio source to update `MediaItem` in `queue` for show on `notification`? In the `_effectiveSequence` at line `500`, inside `map`, i `print` `source.duration`, but it is `null`.

**Comment by ryanheise** [comment]
Are you talking about the playlist example in this plugin, is the playlist in the audio service plugin? It sounds like the latter.

**Comment by phamconganh** [comment]
@ryanheise sorry, i missing. `example playlist` in the `audio service`. But I want to know how get `duration` in the current `AudioSource` in the `AudioPlayer` to update the `notification` duration.

**Comment by ryanheise** [comment]
That seems not relevant to the issue you are commenting on.

**Comment by phamconganh** [comment]
@ryanheise Tks for your quick reply. So, should I open a new issue? I just want to know how get `duration` of item in the playlist.

**Comment by ryanheise** [comment]
I have written a document in this repository called [CONTRIBUTING.md](https://github.com/ryanheise/just_audio/blob/master/CONTRIBUTING.md) explaining how to post issues, how to ask questions, etc.

**Comment by dius00** [comment]
Just bumping this. I apologize for barging in; I am currently quite interested in this feature and if possible I would like to know currently where it stands in the backlog of other issues and/or if this has been shelved for now. Thanks!

**Comment by ryanheise** [comment]
Are you dealing with audio files that your app knows in advance or with arbitrary urls inputted by the user?
